integration-tests/suites/runtime_config_file.go

```diff
@@ -0,0 +1,172 @@
+// Writes various things to the file for runtime configuration and checks
+// the config introspection endpoint to make sure that the configuration is
+// correct.
+func (s *RuntimeConfigFileTestSuite) SetupSuite() {
+	s.RegisterCleanup("external-connection")
+
+	s.StartContainerStats()
+
+	normalizedIp := "255.255.255.255"
+	externalIp := "8.8.8.8"
+	serverPort := 53
+	externalUrl := fmt.Sprintf("http://%s:%d", externalIp, serverPort)
+	image_store := config.Images()
+	containerID, err := s.Executor().StartContainer(
+		config.ContainerStartConfig{
+			Name:    "external-connection",
+			Image:   image_store.QaImageByKey("qa-alpine-curl"),
+			Command: []string{"sh", "-c", "while true; do curl " + externalUrl + "; sleep 1; done"},
+		})
+	s.Require().NoError(err)
+	clientContainer := common.ContainerShortID(containerID)
+
+	collectorOptions := collector.StartupOptions{
+		Env: map[string]string{
+			"ROX_AFTERGLOW_PERIOD":               "6",
+			"ROX_COLLECTOR_INTROSPECTION_ENABLE": "true",
+		},
+	}
+
+	activeNormalizedConnection := types.NetworkInfo{
+		LocalAddress:   "",
+		RemoteAddress:  fmt.Sprintf("%s:%d", normalizedIp, serverPort),
+		Role:           "ROLE_CLIENT",
+		SocketFamily:   "SOCKET_FAMILY_UNKNOWN",
+		CloseTimestamp: types.NilTimestamp,
+	}
+
+	activeUnnormalizedConnection := types.NetworkInfo{
+		LocalAddress:   "",
+		RemoteAddress:  fmt.Sprintf("%s:%d", externalIp, serverPort),
+		Role:           "ROLE_CLIENT",
+		SocketFamily:   "SOCKET_FAMILY_UNKNOWN",
+		CloseTimestamp: types.NilTimestamp,
+	}
+
+	inactiveNormalizedConnection := activeNormalizedConnection
+	inactiveNormalizedConnection.CloseTimestamp = "Not nill time"
+
+	inactiveUnnormalizedConnection := activeUnnormalizedConnection
+	inactiveUnnormalizedConnection.CloseTimestamp = "Not nill time"
+
+	runtimeConfigDir := "/tmp/collector-test"
+	runtimeConfigFile := filepath.Join(runtimeConfigDir, "/runtime_config.yaml")
+	collectorIP := "localhost"
+
+	s.createDirectory(runtimeConfigDir)
+	s.deleteFile(runtimeConfigFile)
+	s.StartCollector(false, &collectorOptions)
+
+	// The runtime config file was deleted before starting collector so there should not be any config
+	assert.AssertNoRuntimeConfig(s.T(), collectorIP)
+	// Since there is no config the default is used, which means external IPs is disabled and we should
+	// expect a normalized connection
+	expectedConnections := []types.NetworkInfo{activeNormalizedConnection}
+	connectionSuccess := s.Sensor().ExpectSameElementsConnections(s.T(), clientContainer, 10*time.Second, expectedConnections...)
+	s.Require().True(connectionSuccess)
+
+	s.setExternalIpsEnable(runtimeConfigFile, true)
+	assert.AssertExternalIps(s.T(), true, collectorIP)
+	// When external IPs is enabled the normalized connection is reported as inactive and the unnormalized connection
+	// is reported as active
+	expectedConnections = append(expectedConnections, activeUnnormalizedConnection, inactiveNormalizedConnection)
```

Comment:
Shouldn't we be cleaning `expectedConnections` between each test? Isn't this just saying "you should see the previous ones and these ones"? How does that work when we turn external IPs back off?

Reply:
I don't think we should clean `expectConnections` the mock server does not clear out the connections that it sees after asserts, so `expectConnections` should not be cleared out either.

> Isn't this just saying "you should see the previous ones and these ones"? 
Yes

 > How does that work when we turn external IPs back off?
 When external IPs is turned back off the connection with the unnormalized connection is reported closed and the normalized connection is reported open. We get two reported connection events in that case.